src/main/java/step3/domain/WinningCountMap.java

```diff
@@ -0,0 +1,49 @@
+package step3.domain;
+
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toMap;
+
+public class WinningCountMap {
+
+    private final Map<Integer, PrizeCalculator> winnerCounterMap;
```

Comment:
PrizeCalculator 내부에 WinningType, winnerCount가 존재하고 있습니다.

현재 Map의 Key로 사용되는 Integer는 matchCount로 보이네요.

WinningType enum 내부 포함된 값으로 보여 Key가 크게 유의미하다는 생각이 들진 않아요.

아래와 같이 Map을 사용하는건 어떨까요?

```suggestion
    private final Map<WinningType, Integer> winnerCounterMap;
```

> 이 부분은 반영하지 않으셔도 됩니다. 의견만 남겨주시면 참고하도록 할게요

Reply:
만들고 나니까 이게 꼭 필요한가... 싶은 클래스가 `PrizeCalculator` 였는데요,
바로 피드백으로 나오네요 ㅎㅎ;;
이 `Map` 에서 value로 사용하지 않으면 `PrizeCalculator`는 더 이상 쓸모가 없습니다.
`PrizeCalculator` 자체를 제거하고, 제안 주신 방향으로 리팩토링 진행했습니다.